decidim-core/app/resolvers/decidim/core/metric_resolver.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Core
+    # A GraphQL resolver to handle `count` and `metric` queries
+    class MetricResolver
+      attr_reader :name
+
+      def initialize(name, organization)
+        @name = name
+        @organization = organization
+        @group_by = :day
+        @counter_field = :cumulative
+      end
+
+      def count
+        metric_scope.max.try(:last) || 0
+      end
+
+      def history
+        metric_scope
+      end
+
+      private
+
+      def metric_scope
+        Decidim::Metric
+          .where(metric_type: name, organization: organization)
+          .group(group_by)
+          .order("#{group_by} DESC")
+          .limit(60)
+          .sum(counter_field)
```

Comment:
Why do we need to group and sum?

Reply:
As an example, Proposals' bar charts will be grouped by category. That means for each category and day (an other params) we'll have one row. So, it's needed to group them with the `group_by` field.

In numbers, for `01/08/18` date we have a total of 5 proposals (showed in HOME charts), but 2 of them are related to `Category ID:1` and the other 3 to `Category ID:2`

```
|      day | category_id | cumulative | quantity |  ...
| 01/08/18 |           1 |          2 |        1 |
| 01/08/18 |           2 |          3 |        0 |
```

Reply:
I'd have created two different metrics, one at organization level (aggregating all the values when  writing the metric, this way we don't need to calculate when reading) and one at the category level, but I guess it's OK.